### PR TITLE
Introduced 'user-agent' header

### LINKED ISSUE #1037: Add a customizable `user-agent` header to the HTTP request sent
When the plugin communicates with ES, the user-agent is manticore's, given that it the library used to communicate to Elasticsearch.

As, lets properly identify logstash by setting the user-agent when making HTTP requests to ES with the following format:

`logstash/7.14.1 (OS=Linux-5.4.0-84-generic-amd64; JVM=AdoptOpenJDK-11.0.11) logstash-output-elasticsearch/11.0.1`

### PR DESCRIPTION
**Pull request by andsel**
## Release notes

Feat: add `user-agent` header passed to the Elasticsearch HTTP connection

## What does this PR do?

Pass an option to the underlying `Manticore` library containing the `user-agent` header. That header contains:
- Logstah version
- plugin version, asking to the `Gem` library
- OS name, version and CPU architechture
- JVM vendor name and version

Example:
```
logstash/7.14.1 (OS=Linux-5.4.0-84-generic-amd64; JVM=AdoptOpenJDK-11.0.11) logstash-output-elasticsearch/11.0.1
```

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] ~~I have made corresponding change to the default configuration files (and/or docker env variables)~~
- [x] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [ ]

## Related issues

- Closes #1037

### CODE REVIEW

**Comment by andsel** [comment]
The red CI on 
```
INTEGRATION=true
ELASTIC_STACK_VERSION=8.x
SNAPSHOT=true
```
seems not related to this PR, it happens also on `master` branch

**Review by roaksoax — CHANGES_REQUESTED** [review]
Hi Andrea,

Thanks for adding the ability to set the user-agent. However, the intention of this is NOT to add a customizable user-agent. Rather, the intention of this is that Logstash doesn't inhering the user agent for the library (e.g. manticore) and sets its information always.

As such, we need to change this in such a way that logstash sets the suer-agent to something similar to:

Logstash/7.x.y (logstash-output-elasticsearch/a.b.c)

Ideally, this should also include the platform. Lets further discuss this.

File: spec/unit/outputs/elasticsearch/http_client_spec.rb
Comment (yaauie) [inline spec/unit/outputs/elasticsearch/http_client_spec.rb:310]: It may, however, be better to let WEBrick bind to a random _available_ port instead of guessing one at random.

If we specify `:Port => 0`, then the resulting `WEBrick::HTTPServer` will bind to an _available_ port and its `config` will include and a `:Port` attribute containing the actual port that was bound.

(source: https://github.com/ruby/webrick/blob/v1.7.0/lib/webrick/server.rb#L112-L114)

```suggestion
            @server = WEBrick::HTTPServer.new :Port => 0, :DocumentRoot => ".",
                     :Logger => Cabin::Channel.get, # silence WEBrick logging
                     :StartCallback => Proc.new {
                           queue.push("started")
                         }
            @port = @server.config[:Port]
```

File: spec/unit/outputs/elasticsearch/http_client_spec.rb
Comment (yaauie) [inline spec/unit/outputs/elasticsearch/http_client_spec.rb:354]: nit: we don't have any specs on `Client#prepare_user_agent`, so ensuring that the user agent we received includes the prepared user agent doesn't sufficiently safeguard us (for example, if `Client#prepare_user_agent` returned an empty string, _any_ string would match).

I would prefer that we set more refined expectations here.

```suggestion
        transmitted_user_agent = request.header['user-agent']
        
        expect(transmitted_user_agent).to include("Logstash/#{LOGSTASH_VERSION}")
        expect(transmitted_user_agent).to match(%r{logstash-output-elasticsearch/[0-9.]+}")
```

**Review by yaauie — COMMENTED** [review]
I've added a few nitpicks, but the implementation looks good.

File: spec/unit/outputs/elasticsearch/http_client_spec.rb
Comment (andsel) [reply spec/unit/outputs/elasticsearch/http_client_spec.rb]: This was solved grabbing the port from the HTTP server. The HTTP server binds to the first free port and the testing code retrieve that port

**Review by roaksoax — APPROVED** [review]
From my perspective this LGTM! Not sure if @yaauie needs another review?

**Review by yaauie — APPROVED** [review]
LGTM 🌮

**Pull request merged**
